pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -116,6 +116,40 @@ func (admitter *PlanAdmitter) validateWarmMigrations() error {
 	return nil
 }
 
+func (admitter *PlanAdmitter) validateLUKS() error {
+	hasLUKS := false
+	for _, vm := range admitter.plan.Spec.VMs {
+		if vm.LUKS.Name != "" {
+			hasLUKS = true
+		}
+	}
+	if !hasLUKS {
+		return nil
+	}
+
+	providerType := admitter.sourceProvider.Type()
+	if providerType != api.VSphere && providerType != api.Ova {
+		err := liberr.New("migration of encrypted disks is not supported for this migration")
+		log.Error(err, "Provider type (non-VSphere & non-OVA) does not support LUKS")
+		return err
+	}
+
+	el9, el9Err := admitter.plan.VSphereUsesEl9VirtV2v()
+	if el9Err != nil {
+		log.Error(el9Err, "Could not analyze plan, failing")
+		return el9Err
+	}
+	if el9 {
+		// LUKS is optional when EL9 virt-v2v image is in use
+		log.Info("LUKS secret is optional when EL9 virt-v2v image is in use, passing")
+	} else {
+		err := liberr.New("migration of encrypted disks is not supported for this migration")
```

Comment:
need to change the message here with a more informative one

Reply:
please revisit